src/ports/postgres/modules/convex/mlp_igd.py_in

```diff
@@ -667,7 +678,8 @@ def _validate_dependent_var(source_table, dependent_varname,
         if is_classification:
             # Currently, classification doesn't accept an
             # array for dep type in IGD
-            _assert("[]" not in expr_type and expr_type in classification_types,
+            _assert(("[]" in expr_type and expr_type[:-2] in int_types) \
```

Comment:
- One-hot encoding is normally supposed to be an integer array, hence was checking only for int type. But, that assumption might be too strict, so will make changes to allow for other numeric types too.
- Good catch. It might be a good idea to disallow anything greater than a 1-D array. Will make an error check for the same.